### DISCUSSION TITLE
Property setter

### DISCUSSION THREAD

**Discussion by rcerljenko** [spatie/laravel-settings#112] [Q&A]
Hi,

How can we achieve a "setter" logic like default Laravel model has? E.g.:
```php
public function setTitleAttribute(string $value): void
{
	$this->attributes['title'] = preg_replace('/\s+/', ' ', $value);
}
```

Cheers!

**Comment by rubenvanassche** [comment]
Hi @rcerljenko,

I wouldn't do this using a setter within the Settings class but just do it before inserting/update a setting. If you do want to group this functionality, you could possibly add a custom cast to achieve this.
